menu.go

```diff
@@ -52,15 +52,17 @@ func menu() int16 {
 	for {
 		pressed, _ := buttons.ReadInput()
 
-		if released && buttons.Pins[shifter.BUTTON_UP].Get() && selected > 0 {
-			selected--
+		if released && buttons.Pins[shifter.BUTTON_UP].Get(){
+			prevSel := selected
```

Comment:
I think it's better if we pre-allocate/define the  variable outside the loop (at the same time as `selected` instead to create it each time.
Also, maybe call it `prevSelected` to keep the parallelism with `selected`

what do you thing?